feat(ras-acc): add notice styles to newspack ui

chickenn00dle opened the pull request:
### All Submissions:

* [ ] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/trunk/.github/CONTRIBUTING.md)?
* [ ] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [ ] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request, and the reason for such changes. -->

Closes https://app.asana.com/0/1207720049982626/1207546771559780/f

This PR adds notice styles to newspack ui along with a notices section to the ui demo:
![Screenshot 2024-07-10 at 12 57 53](https://github.com/Automattic/newspack-plugin/assets/17905991/8435fb71-4e62-41fd-bd4d-161670fc1e1b)

### How to test the changes in this Pull Request:

1. With this branch checked out access the ui demo by adding the following url param to your local dev: `?ui-demo` (you must be logged in as admin)
2. Scroll down to the new notices section and confirm it appears as pictured above

### Other information:

* [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully ran tests with your changes locally?

<!-- Mark completed items with an [x] -->

chickenn00dle committed:
feat(ras-acc): add notices to newspack ui

laurelfulford reviewed (changes requested):
Thanks for tackling this @chickenn00dle! I went a little nuts with comparing it to the Figma file, so if any of this has been discussed elsewhere, please disregard!

* The font size in the Figma file is `14px` (`--newspack-ui-font-size-xs`) -- it's picking up the `16px` base, so it just needs an override for regular text.
* The border radius should be `3px` (`--newspack-ui-border-radius-xs`)
* The border width on the left should be `3px`.
* I think the "info" SVG scale is a little off -- it should be a `16px` icon inside of a `24px` space, but it looks to be ~`18`, so the scale seems a little off compared to the checkmark. 
* The SVGs in the notices are shade darker than the borders, so they should be:
     * Error: border: `--newspack-ui-color-error-50`; icon: `--newspack-ui-color-error-60`
     * Success: border `--newspack-ui-color-success-50`; icon: `--newspack-ui-color-success-60`
     * Info: border : `--newspack-ui-color-warning-30`; icon: `--newspack-ui-color-warning-40`
     * Edited to add: The grey one in the mockup looks to use `--newspack-ui-color-neutral-90` for both.
* The padding looks a little loose -- it ends up being 20px 32px, but it looks like it should be 16px all around (with the border factored in -- so the icon is 13px from the border edge, but 16px from the total box size). 

Just let me know if you have any questions about any of this and apologies for being such a pain! I'm happy to help make tweaks since I'm the one making a stink about pixels here, just let me know 🙂

chickenn00dle committed:
fix: various feedback

chickenn00dle commented:
Thanks for the feedback @laurelfulford!

I've addressed most of your feedback in https://github.com/Automattic/newspack-plugin/pull/3235/commits/10290d89af0d050e7d995c0084518f9b4c75f218

There was just one point I didn't make changes for:

> I think the "info" SVG scale is a little off -- it should be a 16px icon inside of a 24px space, but it looks to be ~18, so the scale seems a little off compared to the checkmark.

I tried exporting the info icon from Figma as an svg, then using that svg code, but for some reason the circle around the `i` wouldn't render in the DOM. I'm not sure if there is an easy tweak we can make to the existing svg code to get the icon to be exactly 16px within the svg, but I should note that the info icon I used is from [`@wordpress/icons` library](https://wordpress.github.io/gutenberg/?path=/story/icons-icon--library) and the sizing is the same as the `error` Notice icon.

In any case, if this is a blocker to getting this out the door, I'd love it if you could provide me with a snippet of `svg` code for the info icon as I'm having trouble getting the exact sizing for it on my end.

Thanks!

laurelfulford reviewed (approved):
Thanks for making all of these changes, @chickenn00dle! I think it's looking really good! 🙌 

For the SVG, I think I got myself too into the weeds there 😅 I ended up tweaking a version of the icon to turn the outline into stroke, and it ended up being 17.5px across, not the 16px that I though it was (that was the centre line of the stroke). Sorry for the wild goose chase there! I think it's fine to leave as is. 🚢
